logbook-servlet/src/test/java/org/zalando/logbook/servlet/AsyncDispatchTest.java

```diff
@@ -3,148 +3,106 @@
 
 /**
  * Verifies that {@link LogbookFilter} handles {@link DispatcherType#ASYNC} correctly.
  */
+@SpringBootTest(webEnvironment = DEFINED_PORT)
+@EnableAutoConfiguration(exclude = ErrorMvcAutoConfiguration.class)
 final class AsyncDispatchTest {
 
-    private final HttpLogFormatter formatter = spy(new ForwardingHttpLogFormatter(new DefaultHttpLogFormatter()));
-    private final HttpLogWriter writer = mock(HttpLogWriter.class);
-
-    private final MockMvc mvc = MockMvcBuilders
-            .standaloneSetup(new ExampleController())
-            .addFilter(new LogbookFilter(Logbook.builder()
-                    .strategy(new Strategy() {
-                        @Override
-                        public HttpRequest process(final HttpRequest request) throws IOException {
-                            return request.withBody().withBody().withoutBody().withBody();
-                        }
-
-                        @Override
-                        public void write(final Precorrelation precorrelation, final HttpRequest request,
-                                final Sink sink) throws IOException {
-
-                            request.withoutBody().withBody();
-                            sink.write(precorrelation, request);
-                        }
-
-                        @Override
-                        public HttpResponse process(final HttpRequest request, final HttpResponse response) throws IOException {
-                            return response.withBody().withBody().withoutBody().withBody();
-                        }
-
-                        @Override
-                        public void write(final Correlation correlation, final HttpRequest request,
-                                final HttpResponse response, final Sink sink) throws IOException {
-
-                            response.withoutBody().withBody();
-                            sink.write(correlation, request, response);
-                        }
-                    })
-                    .sink(new DefaultSink(formatter, writer))
-                    .build()))
-            .build();
+    @Configuration
+    @Import(ExampleController.class)
+    static class TestConfiguration {
+
+        @Bean
+        public Logbook logbook(final HttpLogWriter writer) {
+            return Logbook.builder()
+                    .sink(new DefaultSink(
+                            new DefaultHttpLogFormatter(), writer))
+                    .build();
+        }
+
+
+        @Bean
+        @SuppressWarnings({"rawtypes", "unchecked"}) // as of Spring Boot 2.x
+        public FilterRegistrationBean logbookFilter(final Logbook logbook) {
+            final FilterRegistrationBean registration =
+                    new FilterRegistrationBean(new LogbookFilter(logbook));
+            registration.setDispatcherTypes(REQUEST, ASYNC);
+            return registration;
+        }
+
+    }
+
+    @MockBean
+    private HttpLogWriter writer;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/app/whiskeysierra/logbook/pullRequest?prid=4340480)